core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -548,7 +552,8 @@ private static MetadataUpdate readSetSnapshotRef(JsonNode node) {
 
   private static MetadataUpdate readRemoveSnapshotRef(JsonNode node) {
     String refName = JsonUtil.getString(REF_NAME, node);
-    return new MetadataUpdate.RemoveSnapshotRef(refName);
+    boolean purge = JsonUtil.getBool(PURGE, node);
```

Comment:
older clients won't send this flag, so we shouldn't be failing here

Reply:
Updated. The new logic sets `true` by default for backward compatibility.